flink-jpmml-scala/src/main/scala/io/radicalbit/flink/pmml/scala/api/converter/VectorConverter.scala

```diff
@@ -75,9 +97,10 @@ private[api] object VectorConverter {
   private def getKeyFromModel(evaluator: Evaluator) =
     evaluator.getActiveFields.map(_.getName.getValue)
 
-  private def toDenseData(sparseVector: SparseVector): Seq[Option[Any]] =
-    (0 to sparseVector.size).map { index =>
-      if (sparseVector.indices.contains(index)) Some(sparseVector(index)) else None
+  def toDenseData(sparseVector: SparseVector): Seq[Option[Any]] =
+    (0 until sparseVector.size).collect {
+      case index if (sparseVector.indices.contains(index)) => Some(sparseVector(index))
+      case _ => None
     }
```

Comment:
In this case I will prefer a map instead of a collect since you're not filtering at all. Moreover, this is a scala tabulate case. But I gave a try on performance and tabulate seemed quite slower than mapping..